src/main/java/com/fasterxml/jackson/core/util/TextBuffer.java

```diff
@@ -536,6 +536,37 @@ public char[] contentsAsArray() throws IOException {
      */
     public double contentsAsDouble(final boolean useFastParser) throws NumberFormatException {
         try {
+            if (_resultString == null) {
+                // Has array been requested? Can make a shortcut, if so:
+                if (_resultArray != null) {
+                    validateFPLength(_resultArray.length);
+                    return NumberInput.parseDouble(_resultArray, useFastParser);
+                } else {
+                    // Do we use shared array?
+                    if (_inputStart >= 0) {
+                        if (_inputLen < 1) {
```

Comment:
Similarly, decoder has already confirmed this won't be the case, no need to verify.

Reply:
when I removed this I got a NPE in a new test (empty TextBuffer)

Reply:
Right but we are not building a general purpose functionality: Parsers will never call (and are specifically not to ever call) method without populating buffer with lexically valid number.